portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -32,11 +35,30 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the comment id to be deleted.
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+    
+    // Get the comment id and email to be deleted.
     long id = Long.parseLong(request.getParameter("id"));
+    String commentEmail = request.getParameter("email");
 
-    Key entityKey = KeyFactory.createKey("Comment", id);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(entityKey);
+    // only delete comment if current user created the comment
+    if (commentEmail.equals(email)) {
+      Key entityKey = KeyFactory.createKey("Comment", id);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.delete(entityKey);
+    }
+
+    String json = "{\"success\": ";
+    if (commentEmail.equals(email)) {
+      json += "\"true\"";
+    } else {
+      json += "\"false\"";
+    }
+    json += "}";
```

Comment:
This absolutely works, but I would recommend using string formatting to handle this:

```java
  boolean value = true;
  String sentence = String.format("value is %b", value);
```